src/lib/plugins/code/analysis.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  analyzeFolders,
+  emitter as codeEmitter,
+  AnalysisSeverity,
+} from '@snyk/code-client';
+import { Log, ReportingDescriptor, Result } from 'sarif';
+import { SEVERITY } from '../../snyk-test/legacy';
+import { api } from '../../api-token';
+import * as config from '../../config';
+import spinner = require('../../spinner');
+import { Options } from '../../types';
+import { FeatureNotSupportedBySnykCodeError } from './errors/unsupported-feature-snyk-code-error';
+
+export async function getCodeAnalysisAndParseResults(
+  root: string,
+  options: Options,
+): Promise<Log> {
+  let currentLabel = '';
+  await spinner.clearAll();
+  codeEmitter.on('scanFilesProgress', async (processed: number) => {
+    const spinnerLbl = `Indexed ${processed} files`;
+    spinner.clear<void>(currentLabel)();
+    currentLabel = spinnerLbl;
+    await spinner(spinnerLbl);
+  });
+  codeEmitter.on(
+    'uploadBundleProgress',
+    async (processed: number, total: number) => {
+      const spinnerLbl = `Upload bundle progress: ${processed}/${total}`;
+      spinner.clear<void>(currentLabel)();
+      currentLabel = spinnerLbl;
+      await spinner(spinnerLbl);
+    },
+  );
+  codeEmitter.on('analyseProgress', async (data: any) => {
+    const spinnerLbl = `Analysis ${data.status}: ${Math.round(
+      data.progress * 100,
+    )}%`;
+    spinner.clear<void>(currentLabel)();
+    currentLabel = spinnerLbl;
+    await spinner(spinnerLbl);
+  });
+
+  codeEmitter.on('sendError', (error) => {
+    throw error;
+  });
+  const codeAnalysis = await getCodeAnalysis(root, options);
+  spinner.clearAll();
+  return parseSecurityResults(codeAnalysis);
+}
+
+async function getCodeAnalysis(root: string, options: Options): Promise<Log> {
+  const baseURL = config.CODE_CLIENT_PROXY_URL;
+  const sessionToken = api() || '';
+
+  const severity = options.severityThreshold
+    ? severityToAnalysisSeverity(options.severityThreshold)
+    : AnalysisSeverity.info;
+  const paths: string[] = [root];
+  const sarif = true;
+
+  const result = await analyzeFolders({
+    baseURL,
+    sessionToken,
+    severity,
+    paths,
+    sarif,
+  });
+
+  return result.sarifResults!;
+}
+
+function severityToAnalysisSeverity(severity: SEVERITY): AnalysisSeverity {
+  if (severity === SEVERITY.CRITICAL) {
+    throw new FeatureNotSupportedBySnykCodeError(SEVERITY.CRITICAL);
+  }
+  const severityLevel = {
+    low: 1,
+    medium: 2,
+    high: 3,
+  };
+  return severityLevel[severity];
+}
+
+function parseSecurityResults(codeAnalysis: Log): Log {
```

Comment:
Had to read the code in order to understand what the function does... maybe we can rename?
```suggestion
function filterNonSecurityResults(codeAnalysis: Log): Log {
```